app.js

```diff
@@ -0,0 +1,98 @@
+var app = angular.module('PathOfDamage', []);
+app.controller('Damage', function ($scope) {
+  $scope.hits = [100, 500, 1000, 2000, 3000, 4000, 5000, 7500, 10000];
+
+  $scope.buffs = {
+    total: 0,
+    values: []
+  };
+  $scope.shifts = {
+    total: 0,
+    values: []
+  };
+  $scope.damageReductions = {
+    total: 0,
+    values: []
+  };
+  $scope.flatTaken = {
+    total: 0,
+    values: []
+  };
+  $scope.reducedTaken = {
+    total: 0,
+    values: []
+  };
+  $scope.lessTaken = {
+    total: 0,
+    values: []
+  };
+
+  $scope.armor = "";
+  $scope.charges = "";
+  $scope.resistance = 75;
+
+  $scope.add = function (table) {
+    var lastEntry = table.values[table.values.length - 1];
+    if (!lastEntry || lastEntry.name !== "" || lastEntry.value !== "") {
+      table.values.push({
+        enabled: true,
+        name: "",
+        value: ""
+      });
+    }
+    $scope.updateTotal(table);
+  };
+
+  $scope.delete = function (table, index) {
+    if (index !== table.length - 1) {
+      table.splice(index, 1);
+    }
+    $scope.updateTotal(table);
+  };
+
+  $scope.updateTotal = function (table) {
+    table.total = 0;
+    for (var i=0; i<table.values.length; i++) {
```

Comment:
I read http://stackoverflow.com/questions/9329446/for-each-over-an-array-in-javascript and it seemed like the forEach was less readable, because it basically passes in a function to apply to each iterated value. Is it better practice?